src/main/java/lotto/domain/LottoTicket.java

```diff
@@ -20,23 +21,23 @@ public LottoTicket(List<Integer> lottoTicket) {
     }
 
     public Rank matchRank(final WinningNumber winningNumber) {
-        int matchCount = getMatchCount(winningNumber);
+        long matchCount = getMatchCount(winningNumber);
         boolean hasBonusNumber = isBonusNumber(winningNumber, matchCount);
 
         return Rank.valudOf(matchCount, hasBonusNumber);
     }
 
-    public int getMatchCount(final WinningNumber winningNumbers) {
-        return IntStream.range(NUMBER_ZERO, lottoTicket.size())
+    public long getMatchCount(final WinningNumber winningNumbers) {
+        return LongStream.range(NUMBER_ZERO, lottoTicket.size())
                 .map(number -> getContainsLottoNumber(winningNumbers, number))
                 .sum();
     }
 
-    private int getContainsLottoNumber(final WinningNumber winningNumbers, int number) {
-        return winningNumbers.isContainsLottoNumber(getLottoTicketNumber(number)) ? NUMBER_ONE : NUMBER_ZERO;
+    private long getContainsLottoNumber(final WinningNumber winningNumbers, long number) {
+        return winningNumbers.isContainsLottoNumber(getLottoTicketNumber((int) number)) ? new Count(NUMBER_ZERO).update() : new Count(NUMBER_ZERO).getCount();
```

Comment:
위에도 남겨두었지만

```suggestion
        return winningNumbers.isContainsLottoNumber(getLottoTicketNumber((int) number)) ? 1 : 0;
```

와 같이 쓰면 어떨까요?